lib/rbeapi/api/routemaps.rb

```diff
@@ -43,63 +43,439 @@ module Api
+      ##
+      # getall returns a collection of routemap resource hashes from the nodes
+      # running configuration. The routemap resource collection hash is keyed
+      # by the unique routemap name.
+      #
+      # @example
+      #   [
+      #     <test:10>: {
+      #       action: <string>,
+      #       seqno: <integer>,
+      #       match_rules: <array>,
+      #       set_rules: <array>,
+      #       continue: <integer>
+      #       description: <string>
+      #     },
+      #     ...
+      #   ]
+      #
+      # @return [Hash<Symbol, Object>] returns a hash that represents the
+      #   entire routemap collection from the nodes running configuration.  If
+      #   there are no routemap names configured, this method will return an
+      #   empty hash.
+      def getall
+        routemaps = config.scan(/(?<=^route-map\s)[^\s]+/)
+        return nil unless routemaps
+        response = {}
+        routemaps.each do |name|
+          entries = entries_regex(name)
+          response[name] = parse_entries(entries)
+        end
+        response
+      end
+
+      ##
+      # entries regex scans the config for the specified route-name. The method
+      #   reduces duplication.
+      #
+      def entries_regex(name)
+        config.scan(/^route-map\s#{name}\s.+$/)
+      end
+      private :entries_regex
+
+      ##
+      # parse entries is a private method to get the routemap rules.
+      #
+      # @return [Hash<Symbol, Object>] returns a hash that represents the
+      #   rules for routemaps from the nodes running configuration.  If
+      #   there are no routemaps configured, this method will return an empty
+      #    hash.
+      #
+      def parse_entries(entries)
+        entries.each_with_object([]) do |rm|
           mdata = /route-map\s(.+)\s(.+)\s(\d+)$/.match(rm)
           rules = get_block(rm)
-          rule_hsh = { 'action' => mdata[2], 'seqno' => mdata[3],
-                       'match_rules' => [], 'set_rules' => [],
-                       'continue_rules' => [] }
-
-          parsed = rules.split("\n").each_with_object({}) do |rule, hsh|
-            mdata = /\s{3}(\w+)\s/.match(rule)
-            case mdata.nil? ? nil : mdata[1]
-            when 'match'
-              hsh['match_rules'] = [] unless hsh.include?('match')
-              hsh['match_rules'] << rule.strip
-            when 'set'
-              hsh['set_rules'] = [] unless hsh.include?('set')
-              hsh['set_rules'] << rule.strip
-            when 'continue'
-              hsh['continue_rules'] = [] unless hsh.include?('continue')
-              hsh['continue_rules'] << rule.strip
+          rule_hsh = { action: mdata[2], seqno: mdata[3].to_i }
+          unless rules.nil?
+            rules.split("\n").each_with_object({}) do |rule|
+              parse_rule(rule, rule_hsh)
             end
           end
-          rule_hsh.update(parsed)
-          arry << rule_hsh
+          return rule_hsh
```

Comment:
rule_hsh is only available in the loop. I can think about another way to write this if you don't like it.